### ISSUE TITLE
pullrequest-builder-plugin doesn't work as expected with Jenkins Pipeline script.

### ISSUE THREAD

**Issue by pavanreddy42** [nishio-dens/bitbucket-pullrequest-builder-plugin#150]
Hello all,  I am using the below configuration in my Jenkins pipeline file but my Jenkins server send the inprogress build status to bit bucket using the CI key but it is not updating the status once the build is finished. when I tried using the same plugin in free style project it send the build status using the CI name and updates the build status also once the build is finished. 
  triggers{
        bitbucketpr(projectPath:'################################',
        cron:'* * * * *',
        credentialsId:'##################',
        username:'########',
        password:'#########',
        repositoryOwner:'#####',
        repositoryName:'#####',
        branchesFilter:'',
        branchesFilterBySCMIncludes:false,
        ciKey:'',
        ciName:'',
        ciSkipPhrases:'',
        checkDestinationCommit:true,
        approveIfSuccess:true,
        cancelOutdatedJobs:true,
        commentTrigger:'test this please')
    } 
![capture](https://user-images.githubusercontent.com/38340284/38717257-aec24752-3e9a-11e8-9f08-413c600c7de1.PNG)

**Comment by Aamir010** [comment]
Is there any update on this issue. Same error with me when using Jenkinsfile.

**Comment by damovsky** [comment]
sorry so say this...but I'm not able to continue to work on this project alone, I need help (I have a newborn baby and not enough time to work on this project).

**Comment by damovsky** [comment]
I'm looking for a new maintainer for this project. Any volunteer ?

**Comment by CodeMonk** [comment]
@abhilashshettigar , @Aamir010 , @pavanreddy42 :I think this is the same thing as #155  - Do you have the ability to test with tip to see if your problem goes away?

If not, I'm planning on pushing out a release in the next week, once I figure out how :)

**Comment by CodeMonk** [comment]
Duplicate of #166

**Comment by pratapsingh** [comment]
I am facing the same issue and I am sure its not due to the rate limiting as I am using Bitbucket enterprise. I have also checked with Bitbucket support team that we are having enough bandwidth avaialble so we can not hit rate limit. 

It works fine for some time then it does not run the build untill I restart the jenkins. After restarting Jenins it again works like charm and after some time same behaviour occurs. So I am assuming it some kind of leak in plugin.
